agent/api/registryauth.go

```diff
@@ -11,7 +11,7 @@
 // express or implied. See the License for the specific language governing
 // permissions and limitations under the License.
 
-package ecr
+package api
```

Comment:
Can you keep this out of the main `api` package?  If you want it more in that path, I'd suggest something like `github.com/aws/amazon-ecs-agent/agent/api/auth` instead.

Reply:
discussed this with @yunhee-l, and since the task-resource branch does a heavy refactor of the api pkg and introduces subpackages, we decided it would be best to keep it in api for now.